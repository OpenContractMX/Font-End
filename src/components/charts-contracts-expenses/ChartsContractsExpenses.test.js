import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { ChartsContractsExpenses } from "./";

jest.mock("recharts", () => {
  const MockResponsiveContainer = (props) => <div {...props} />;

  return {
    ...jest.requireActual("recharts"),
    ResponsiveContainer: MockResponsiveContainer,
  };
});

it("[render content `statistic / ChartsContractsExpenses`]", () => {
  const contractsChars = {
    contracts_number: "0",
    inversion: 0,
    top_ten: [
      {
        id: 0,
        month: 0,
        trimester: 0,
        title: "vacio",
        buyer_name: "vacio",
        date: "0",
        amount: 0,
        currency: "",
      },
    ],
    execution_mean: 0,
    months: {
      1: 0,
      2: 0,
      3: 0,
      4: 0,
      5: 0,
      6: 0,
      7: 0,
      8: 0,
      9: 0,
      10: 0,
      11: 0,
      12: 0,
    },
  };
  const { asFragment } = render(
    <ChartsContractsExpenses contractsChars={contractsChars} />
  );
  expect(asFragment()).toMatchSnapshot();
});
