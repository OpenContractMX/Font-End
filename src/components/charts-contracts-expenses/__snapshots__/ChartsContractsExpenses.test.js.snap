// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[render content \`statistic / ChartsContractsExpenses\`] 1`] = `
<DocumentFragment>
  <section
    class="container__contract-expense"
  >
    <h2
      class="container__contract-expense--title"
    >
      Top 10 contratos en gastos:
    </h2>
    <div
      class="container__contract-expense--expenses-wrap"
    >
      <div
        aspect="2"
        width="100%"
      >
        <div
          class="recharts-wrapper container__contract-expense--expenses-wrap--chart"
          style="position: relative; cursor: default; width: 380px; height: 240px;"
        >
          <svg
            class="recharts-surface"
            height="240"
            version="1.1"
            viewBox="0 0 380 240"
            width="380"
          >
            <defs>
              <clippath
                id="recharts1-clip"
              >
                <rect
                  height="190"
                  width="310"
                  x="60"
                  y="10"
                />
              </clippath>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="190"
                  offset="[object Object]"
                  stroke="#243240"
                  width="310"
                  x="60"
                  x1="60"
                  x2="370"
                  y="10"
                  y1="200"
                  y2="200"
                />
                <line
                  fill="none"
                  height="190"
                  offset="[object Object]"
                  stroke="#243240"
                  width="310"
                  x="60"
                  x1="60"
                  x2="370"
                  y="10"
                  y1="152.5"
                  y2="152.5"
                />
                <line
                  fill="none"
                  height="190"
                  offset="[object Object]"
                  stroke="#243240"
                  width="310"
                  x="60"
                  x1="60"
                  x2="370"
                  y="10"
                  y1="105"
                  y2="105"
                />
                <line
                  fill="none"
                  height="190"
                  offset="[object Object]"
                  stroke="#243240"
                  width="310"
                  x="60"
                  x1="60"
                  x2="370"
                  y="10"
                  y1="57.5"
                  y2="57.5"
                />
                <line
                  fill="none"
                  height="190"
                  offset="[object Object]"
                  stroke="#243240"
                  width="310"
                  x="60"
                  x1="60"
                  x2="370"
                  y="10"
                  y1="10"
                  y2="10"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="310"
                x="60"
                x1="60"
                x2="370"
                y="200"
                y1="200"
                y2="200"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#666"
                    type="category"
                    width="310"
                    x="60"
                    x1="215"
                    x2="215"
                    y="200"
                    y1="206"
                    y2="200"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#1a233a"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    type="category"
                    width="310"
                    x="215"
                    y="208"
                  >
                    <tspan
                      dy="0.71em"
                      x="215"
                    >
                      ID: 1
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="190"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="0"
                x1="60"
                x2="60"
                y="10"
                y1="10"
                y2="200"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="190"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="0"
                    x1="54"
                    x2="60"
                    y="10"
                    y1="200"
                    y2="200"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#1a233a"
                    height="190"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="52"
                    y="200"
                  >
                    <tspan
                      dy="0.355em"
                      x="52"
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="190"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="0"
                    x1="54"
                    x2="60"
                    y="10"
                    y1="152.5"
                    y2="152.5"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#1a233a"
                    height="190"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="52"
                    y="152.5"
                  >
                    <tspan
                      dy="0.355em"
                      x="52"
                    >
                      1
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="190"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="0"
                    x1="54"
                    x2="60"
                    y="10"
                    y1="105"
                    y2="105"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#1a233a"
                    height="190"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="52"
                    y="105"
                  >
                    <tspan
                      dy="0.355em"
                      x="52"
                    >
                      2
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="190"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="0"
                    x1="54"
                    x2="60"
                    y="10"
                    y1="57.5"
                    y2="57.5"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#1a233a"
                    height="190"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="52"
                    y="57.5"
                  >
                    <tspan
                      dy="0.355em"
                      x="52"
                    >
                      3
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="190"
                    orientation="left"
                    stroke="#666"
                    type="number"
                    width="60"
                    x="0"
                    x1="54"
                    x2="60"
                    y="10"
                    y1="10"
                    y2="10"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#1a233a"
                    height="190"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    type="number"
                    width="60"
                    x="52"
                    y="10"
                  >
                    <tspan
                      dy="0.355em"
                      x="52"
                    >
                      4
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-bar"
            >
              <g
                class="recharts-layer recharts-bar-rectangles"
              >
                <g
                  class="recharts-layer"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                </g>
              </g>
            </g>
          </svg>
          <div
            class="recharts-tooltip-wrapper"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap; color: rgb(26, 35, 58);"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              >
                0
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container__contract-expense--wrap"
    />
  </section>
</DocumentFragment>
`;
